### ISSUE TITLE
feat(core): Mixed Precision Policy API

### ISSUE THREAD

**Issue by m96-chan** [m96-chan/PyGPUkit#160]
## Overview

Add configurable mixed precision policies for automatic dtype selection.

## Motivation

Different operations benefit from different precisions:
- MatMul: TF32/FP16/BF16/FP8
- Attention: FP32 for softmax stability
- Normalization: FP32 accumulation
- Embedding: FP16/BF16

## Features

- [ ] Precision policy configuration
- [ ] Per-operation dtype override
- [ ] Automatic loss scaling (for training)
- [ ] Policy presets (speed, quality, balanced)

## Design

```python
from pygpukit import PrecisionPolicy

# Built-in presets
policy = PrecisionPolicy.speed()    # Maximize throughput
policy = PrecisionPolicy.quality()  # Maximize accuracy
policy = PrecisionPolicy.balanced() # Default

# Custom policy
policy = PrecisionPolicy(
    matmul="tf32",
    attention="fp32", 
    norm="fp32",
    embedding="bf16",
)

# Apply to model
model.set_precision_policy(policy)
```

## Related

- PyTorch autocast
- NVIDIA Transformer Engine
